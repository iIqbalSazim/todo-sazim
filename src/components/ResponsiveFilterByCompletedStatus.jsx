import { Button, Center, MediaQuery } from "@mantine/core";
import { COLORS, COMPLETED_STATUS } from "../constants/constant";

const ResponsiveFilterByCompletedStatus = ({
  setCompletedStatusFilter,
  filter,
}) => {
  return (
    <>
      <MediaQuery largerThan={"md"} styles={{ display: "none" }}>
        <Center>
          <Button.Group>
            <Button
              variant={
                filter.status === COMPLETED_STATUS.ALL ? "filled" : "outline"
              }
              color={COLORS.BUTTON}
              onClick={() => setCompletedStatusFilter(COMPLETED_STATUS.ALL)}
            >
              All
            </Button>
            <Button
              variant={
                filter.status === COMPLETED_STATUS.ACTIVE ? "filled" : "outline"
              }
              color={COLORS.BUTTON}
              onClick={() => setCompletedStatusFilter(COMPLETED_STATUS.ACTIVE)}
            >
              Active
            </Button>
            <Button
              variant={
                filter.status === COMPLETED_STATUS.COMPLETED
                  ? "filled"
                  : "outline"
              }
              color={COLORS.BUTTON}
              onClick={() =>
                setCompletedStatusFilter(COMPLETED_STATUS.COMPLETED)
              }
            >
              Completed
            </Button>
          </Button.Group>
        </Center>
      </MediaQuery>
    </>
  );
};

export default ResponsiveFilterByCompletedStatus;
