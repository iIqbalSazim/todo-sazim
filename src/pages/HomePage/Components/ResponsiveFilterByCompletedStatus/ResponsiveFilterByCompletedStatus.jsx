import { Button, Center, MediaQuery } from "@mantine/core";

import { COLORS, COMPLETED_STATUS } from "../../HomePageConstants";

const ResponsiveFilterByCompletedStatus = ({
  setCompletedStatusFilter,
  filter,
}) => {
  return (
    <>
      <MediaQuery largerThan={"md"} styles={{ display: "none" }}>
        <Center>
          <Button.Group>
            <Button
              variant={
                filter.status === COMPLETED_STATUS.ALL ? "filled" : "outline"
              }
              color={COLORS.BUTTON}
              onClick={() => setCompletedStatusFilter(COMPLETED_STATUS.ALL)}
              size="xs"
            >
              All
            </Button>
            <Button
              variant={
                filter.status === COMPLETED_STATUS.ACTIVE ? "filled" : "outline"
              }
              color={COLORS.BUTTON}
              onClick={() => setCompletedStatusFilter(COMPLETED_STATUS.ACTIVE)}
              size="xs"
            >
              Active
            </Button>
            <Button
              variant={
                filter.status === COMPLETED_STATUS.COMPLETED
                  ? "filled"
                  : "outline"
              }
              color={COLORS.BUTTON}
              onClick={() =>
                setCompletedStatusFilter(COMPLETED_STATUS.COMPLETED)
              }
              size="xs"
            >
              Completed
            </Button>
          </Button.Group>
        </Center>
      </MediaQuery>
    </>
  );
};

export default ResponsiveFilterByCompletedStatus;
