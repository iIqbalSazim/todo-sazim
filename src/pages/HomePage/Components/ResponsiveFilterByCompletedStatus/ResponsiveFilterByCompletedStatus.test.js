import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import ResponsiveFilterByCompletedStatus from "./ResponsiveFilterByCompletedStatus";

const mockSetCompletedStatusFilter = jest.fn();

describe("ResponsiveFilterByCompletedStatus", () => {
  test("renders correctly", () => {
    const { container } = render(
      <ResponsiveFilterByCompletedStatus
        setCompletedStatusFilter={mockSetCompletedStatusFilter}
        filter={{}}
      />
    );

    const allButton = screen.getByRole("button", {
      name: /all/i,
    });
    const activeButton = screen.getByRole("button", {
      name: /active/i,
    });
    const completedButton = screen.getByRole("button", {
      name: /completed/i,
    });

    expect(allButton).toBeInTheDocument();
    expect(activeButton).toBeInTheDocument();
    expect(completedButton).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("calls setCompletedStatusFilter with value 'all' when clicked on All button", async () => {
    const { container } = render(
      <ResponsiveFilterByCompletedStatus
        setCompletedStatusFilter={mockSetCompletedStatusFilter}
        filter={{}}
      />
    );

    const allButton = screen.getByRole("button", {
      name: /all/i,
    });

    await userEvent.click(allButton);

    expect(mockSetCompletedStatusFilter).toHaveBeenCalledWith("all");
    expect(container).toMatchSnapshot();
  });

  test("calls setCompletedStatusFilter with value 'active' when clicked on Active button", async () => {
    const { container } = render(
      <ResponsiveFilterByCompletedStatus
        setCompletedStatusFilter={mockSetCompletedStatusFilter}
        filter={{}}
      />
    );

    const active = screen.getByRole("button", {
      name: /active/i,
    });

    await userEvent.click(active);

    expect(mockSetCompletedStatusFilter).toHaveBeenCalledWith("active");
    expect(container).toMatchSnapshot();
  });

  test("calls setCompletedStatusFilter with value 'completed' when clicked on Completed button", async () => {
    const { container } = render(
      <ResponsiveFilterByCompletedStatus
        setCompletedStatusFilter={mockSetCompletedStatusFilter}
        filter={{}}
      />
    );

    const completedButton = screen.getByRole("button", {
      name: /completed/i,
    });

    await userEvent.click(completedButton);

    expect(mockSetCompletedStatusFilter).toHaveBeenCalledWith("completed");
    expect(container).toMatchSnapshot();
  });
});
