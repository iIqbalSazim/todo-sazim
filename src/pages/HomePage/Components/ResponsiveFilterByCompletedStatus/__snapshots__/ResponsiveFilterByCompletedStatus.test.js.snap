// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveFilterByCompletedStatus calls setCompletedStatusFilter with value 'active' when clicked on Active button 1`] = `
<div>
  <div
    class="mantine-Center-root mantine-MediaQuery-media mantine-144vvff"
  >
    <div
      class="mantine-ButtonGroup-root mantine-1abrqr0"
    >
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            All
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Active
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Completed
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`ResponsiveFilterByCompletedStatus calls setCompletedStatusFilter with value 'all' when clicked on All button 1`] = `
<div>
  <div
    class="mantine-Center-root mantine-MediaQuery-media mantine-144vvff"
  >
    <div
      class="mantine-ButtonGroup-root mantine-1abrqr0"
    >
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            All
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Active
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Completed
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`ResponsiveFilterByCompletedStatus calls setCompletedStatusFilter with value 'completed' when clicked on Completed button 1`] = `
<div>
  <div
    class="mantine-Center-root mantine-MediaQuery-media mantine-144vvff"
  >
    <div
      class="mantine-ButtonGroup-root mantine-1abrqr0"
    >
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            All
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Active
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Completed
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`ResponsiveFilterByCompletedStatus renders correctly 1`] = `
<div>
  <div
    class="mantine-Center-root mantine-MediaQuery-media mantine-144vvff"
  >
    <div
      class="mantine-ButtonGroup-root mantine-1abrqr0"
    >
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            All
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Active
          </span>
        </div>
      </button>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-xzdc54"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Completed
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
